packages/server/graphql/mutations/endRetrospective.ts

```diff
@@ -25,16 +25,123 @@ import generateWholeMeetingSentimentScore from './helpers/generateWholeMeetingSe
 import generateWholeMeetingSummary from './helpers/generateWholeMeetingSummary'
 import handleCompletedStage from './helpers/handleCompletedStage'
 import {IntegrationNotifier} from './helpers/notifications/IntegrationNotifier'
+import publishNotification from '../public/mutations/helpers/publishNotification'
 import removeEmptyTasks from './helpers/removeEmptyTasks'
 import updateQualAIMeetingsCount from './helpers/updateQualAIMeetingsCount'
 import gatherInsights from './helpers/gatherInsights'
+import NotificationKudosReceived from '../../database/types/NotificationKudosReceived'
+import {RawDraftContentState} from 'draft-js'
 
 const getTranscription = async (recallBotId?: string | null) => {
   if (!recallBotId) return
   const manager = new RecallAIServerManager()
   return await manager.getBotTranscript(recallBotId)
 }
 
+const sendKudos = async (meeting: MeetingRetrospective, teamId: string, context: GQLContext) => {
+  const {dataLoader, socketId: mutatorId} = context
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+  const {id: meetingId, disableAnonymity} = meeting
+  const isAnonymous = !disableAnonymity
+  const pg = getKysely()
+  const r = await getRethink()
+
+  const [reflections, team] = await Promise.all([
+    dataLoader.get('retroReflectionsByMeetingId').load(meetingId),
+    dataLoader.get('teams').loadNonNull(teamId)
+  ])
+
+  const {giveKudosWithEmoji, kudosEmojiUnicode, kudosEmoji} = team
+
+  if (!giveKudosWithEmoji || !kudosEmojiUnicode) {
+    return
+  }
+
+  const kudosToInsert: {
+    senderUserId: string
+    receiverUserId: any
+    teamId: string
+    emoji: string
+    emojiUnicode: string
+    isAnonymous: boolean
+    reflectionId: string
+  }[] = []
+
+  const notificationsToInsert: NotificationKudosReceived[] = []
+
+  for (const reflection of reflections) {
+    const {id: reflectionId, content, plaintextContent, creatorId} = reflection
+    const senderUser = await dataLoader.get('users').loadNonNull(creatorId)
+
+    const contentJson = JSON.parse(content) as RawDraftContentState
+
+    if (plaintextContent.includes(kudosEmojiUnicode) && contentJson.entityMap) {
```

Comment:
+1 I think as you've introduced mentions in the reflections, we should also send out normal mention notifications if:
* it wasn't kudos
* the receiver is not in the meeting

Reply:
Make sense, can we make it in next PR https://github.com/ParabolInc/parabol/issues/9328 ?

Reply:
Sure